Source/WebCore/Modules/applepay/paymentrequest/ApplePayPaymentHandler.cpp

```diff
@@ -546,6 +550,35 @@ static ExceptionOr<void> validate(const ApplePayModifier& applePayModifier)
     }
 #endif
 
+#if ENABLE(APPLE_PAY_DEFERRED_PAYMENTS)
+    if (auto& deferredPaymentRequest = applePayModifier.deferredPaymentRequest) {
+        auto& deferredBilling = deferredPaymentRequest->deferredBilling;
+        if (deferredBilling.paymentTiming != ApplePayPaymentTiming::Deferred)
+            return Exception(TypeError, "'deferredBilling' must be a 'deferred' line item."_s);
+        if (!deferredBilling.label)
+            return Exception(TypeError, "Missing label for 'deferredBilling'."_s);
+        if (!isValidDecimalMonetaryValue(deferredBilling.amount) && deferredBilling.type != ApplePayLineItem::Type::Pending)
+            return Exception(TypeError, makeString('"', deferredBilling.amount, "\" is not a valid amount."));
+
+        if (auto& managementURL = deferredPaymentRequest->managementURL; !URL { managementURL }.isValid())
+            return Exception(TypeError, makeString('"', managementURL, "\" is not a valid URL."));
+
+        if (auto& freeCancellationDate = deferredPaymentRequest->freeCancellationDate; std::isnan(freeCancellationDate)) {
+            if (auto& freeCancellationDateTimeZone = deferredPaymentRequest->freeCancellationDateTimeZone; !freeCancellationDateTimeZone.isEmpty())
+                return Exception(TypeError, "Unexpected 'freeCancellationDateTimeZone' when 'freeCancellationDate' is not set."_s);
+        } else if (auto& freeCancellationDateTimeZone = deferredPaymentRequest->freeCancellationDateTimeZone; freeCancellationDateTimeZone.isEmpty())
+            return Exception(TypeError, "Missing 'freeCancellationDateTimeZone' when 'freeCancellationDate' is set."_s);
+        else if (!CFTimeZoneCreateWithName(nullptr, freeCancellationDateTimeZone.createCFString().get(), false))
```

Comment:
This should use `isTimeZoneValid ` rather than `CFTimeZoneCreateWithName`. Hopefully addressed by my other comment.

Reply:
Yeah, and generally we try to avoid Apple-specific stuff like CF outside of platform. (Not that important here considering it is an Apple-specific feature, and a module)